src/uct/ib/dc/accel/dc_mlx5.c

```diff
@@ -551,6 +555,69 @@ static UCS_F_NOINLINE void uct_dc_mlx5_iface_handle_failure(uct_ib_iface_t *ib_i
     uct_ib_mlx5_completion_with_err((void*)cqe, 0);
 }
 
+ucs_status_t uct_dc_mlx5_ep_fc_ctrl(uct_ep_t *tl_ep, unsigned op,
+                                    uct_rc_fc_request_t *req)
+{
+    uintptr_t sender_ep;
+    uct_ib_iface_t *ib_iface;
+    uct_ib_mlx5_base_av_t av;
+    uct_dc_fc_request_t *dc_req;
+    uct_dc_mlx5_ep_t *dc_mlx5_ep;
+    uct_dc_ep_t *dc_ep         = ucs_derived_of(tl_ep, uct_dc_ep_t);
+    uct_dc_mlx5_iface_t *iface = ucs_derived_of(tl_ep->iface,
+                                                uct_dc_mlx5_iface_t);
+    UCT_DC_MLX5_TXQP_DECL(txqp, txwq);
+
+    ucs_assert((sizeof(uint8_t) + sizeof(sender_ep)) <=
+                UCT_IB_MLX5_AV_FULL_SIZE);
+
+    UCT_DC_CHECK_RES(&iface->super, dc_ep);
+    UCT_DC_MLX5_IFACE_TXQP_GET(iface, dc_ep, txqp, txwq);
+
+    if (op == UCT_RC_EP_FC_PURE_GRANT) {
+        ucs_assert(req != NULL);
+        dc_req    = ucs_derived_of(req, uct_dc_fc_request_t);
+        sender_ep = (uintptr_t)dc_req->sender_ep;
+        ib_iface  = &iface->super.super.super;
+
+        /* Note av initialization is copied from exp verbs */
+        av.stat_rate_sl = ib_iface->config.sl; /* (attr->static_rate << 4) | attr->sl */
+        av.fl_mlid      = ib_iface->path_bits[0] & 0x7f;
+
+        /* lid in dc_req is in BE already  */
+        av.rlid         = dc_req->lid | htons(ib_iface->path_bits[0]);
```

Comment:
not sure I understand. lid is only needed in fc_ctrl, which are different in verbs and mlx5 codes. So we are using the lid in the form it is passed to fc_handler function (LE for verbs and BE for mlx5). This way we are avoiding any conversions for both - verbs and mlx5.